lib/mappers/esdn_mapper.py

```diff
@@ -0,0 +1,348 @@
+from akara import logger
+from dplaingestion.utilities import iterify
+from dplaingestion.selector import exists, getprop
+from dplaingestion.mappers.oai_mods_mapper import OAIMODSMapper
+
+
+class EsdnMapper(OAIMODSMapper):
+    def __init__(self, provider_data):
+        super(EsdnMapper, self).__init__(provider_data)
+
+    def map_collection(self):
+        ret_dict = {"collection": getprop(self.provider_data, "collection")}
+
+        prop = self.root_key + "relatedItem"
+
+        if exists(self.provider_data, prop):
+            related_items = iterify(getprop(self.provider_data, prop))
+            host_collections = [item for item in related_items
+                                if item.get("type") == "host" and
+                                item.get("displayLabel") == "Collection"]
+
+            for host_collection in host_collections:
+                title = getprop(host_collection, "titleInfo/title", True)
+                if title:
+                    ret_dict["collection"]["title"] = title
+
+                description = getprop(host_collection, "abstract", True)
+                if description:
+                    ret_dict["collection"]["description"] = description
+
+        self.update_source_resource(ret_dict)
+
+    def map_creator_and_contributor(self):
+        prop = self.root_key + "name"
+        mapped_props = {
+            "creator": [],
+            "contributor": []
+        }
+
+        if exists(self.provider_data, prop):
+            for s in iterify(getprop(self.provider_data, prop)):
+                name = s.get("namePart")
+                if name:
+                    role_terms = []
+                    try:
+                        for r in iterify(s.get("role")):
+                            role_term = r.get("roleTerm")
+                            if isinstance(role_term, dict):
+                                role_terms.append(
+                                        role_term.get("#text").lower())
+                            elif isinstance(role_term, list):
+                                for rt in role_term:
+                                    role_terms.append(rt.lower())
```

Comment:
Thanks for adding the `list` case!

If this is a list of `dict`s, we'll want to find the '#text' property of each, as we do in the `dict` case.

It turns out that I added a function a long time ago to streamline getting '#text' properties, but it's easy to overlook because it's only been used in the Missouri mapper. See `textnode.textnode()`. I think that all of the places in this module where a '#text' property is being accessed could be streamlined with that. (It works on a string _or_ a dict that has a '#text' property.)